pkg/volume/util/operationexecutor/operation_generator.go

```diff
@@ -763,19 +763,21 @@ func (og *operationGenerator) GenerateExpandVolumeFunc(
 				og.recorder.Eventf(pvcWithResizeRequest.PVC, v1.EventTypeWarning, kevents.VolumeResizeFailed, updateErr.Error())
 				return updateErr
 			}
-		}
-
-		// No Cloudprovider resize needed, lets mark resizing as done
-		if !volumePlugin.RequiresFSResize() {
-			glog.V(4).Infof("Controller resizing done for PVC %s", pvcWithResizeRequest.QualifiedName())
-			err := resizeMap.MarkAsResized(pvcWithResizeRequest, newSize)
 
-			if err != nil {
-				glog.Errorf("Error marking pvc %s as resized : %v", pvcWithResizeRequest.QualifiedName(), err)
-				og.recorder.Eventf(pvcWithResizeRequest.PVC, v1.EventTypeWarning, kevents.VolumeResizeFailed, err.Error())
-				return err
+			// No filesystem resize needed, lets mark resizing as done
+			if !volumePlugin.RequiresFSResize() {
```

Comment:
This would prevent marking a PVC as resized if PV already meets requirements of PVC. The reason this check was outside the `if` statement was - in some cases saving a PVC after resizing may fail because of controller crash or something. The crash will cause - `pv.Spec.Capacity` to have correct value but `pvc.Status.Capacity` to have old value. 

The check being outside ensured that - in case we *do* attempt resizing of such a PVC, we also update pvc status without actually performing a resize. 

Also, I do not completely understand, how having the check inside the `if` statement helps here. If `PV` matches the pvc requirement then and volume is of type that requires file system resize then code inside both `if` statements will not execute. That also happens currently as well, so what is the difference?

Reply:
I partly see what we are trying to do here, but as per spec - whether a volume requires file system resize or not is decided by comparing `pvc.Spec.Capacity` and `pvc.Status.Capacity` (pardon my inaccurate usage of field names) and hence if `pv.Spec.Capacity` is greater than requested size then for volume types that requires fs resize, it does not mean that we are done with resizing.

Reply:
My first thought was : since we do not implement fs resize now, when finish i think the logic should be
``` 
  if pvSize.Cmp(newSize) < 0 {
          ......
          if !volumePlugin.RequiresFSResize() {
                .....
           } else {
               // do sth to let kubelet know that fs resize is needed
           }
  }
```
if we put the check `if !volumePlugin.RequiresFSResize() ` out of  `if pvSize.Cmp(newSize) < 0` block, this will cause that we need to always check fs resize and do sth about fs resize  even we do not need to do that
After talking with @gnufied , we may change the check here to `if !plugin.RequiresFSResize() { ... } else {  # check pvc.Status and pv.Spec.Capacity  }`  or something like that when we finish implementing the rest of fs resize